### PR TITLE
Add ConstPowerOfTwo trait

### PR DESCRIPTION
**Pull request by kaidokert**
More c0nstify #58

## Summary by Sourcery

Add a const-compatible power-of-two trait and implement it for primitive integers and FixedUInt, including associated tests and const helper functions.

New Features:
- Introduce a ConstPowerOfTwo trait providing const-friendly power-of-two queries and next-power-of-two operations.
- Implement ConstPowerOfTwo for unsigned primitive integer types and FixedUInt, enabling these operations in const contexts.

Tests:
- Add unit tests for FixedUInt power-of-two detection and next-power-of-two behavior, including overflow handling and const-evaluated usage.

## Summary by CodeRabbit

* **New Features**
  * Added a public power-of-two trait with const-compatible and runtime methods to detect powers of two and compute the next power-of-two (with defined overflow behavior).
  * Implementations provided for primitive unsigned integers and the library's fixed-size unsigned type, plus const-enabled wrappers for compile-time use.
* **Tests**
  * Added comprehensive tests covering edge cases, zero handling, overflow, and const/gated scenarios.

### CODE REVIEW

File: src/fixeduint/power_of_two_impl.rs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+c0nst::c0nst! {
+    impl<T: [c0nst] ConstMachineWord + MachineWord, const N: usize> c0nst ConstPowerOfTwo for FixedUInt<T, N> {
+        fn is_power_of_two(&self) -> bool {
+            // A number is a power of two if it has exactly one bit set
+            // and is not zero
+            if self.is_zero() {
+                return false;
+            }
+            ConstPrimInt::count_ones(*self) == 1u32
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/power_of_two_impl.rs]: done

File: src/fixeduint/power_of_two_impl.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            // Find the position of the highest set bit using leading_zeros
+            // bit_length = BIT_SIZE - leading_zeros
+            let total_bits = Self::BIT_SIZE as u32;
+            let leading = ConstPrimInt::leading_zeros(self);
+            let bits = total_bits - leading;
+            // Check if we can shift 1 left by that many bits
+            if bits >= total_bits {
+                return None;
+            }
+            Some(Self::one() << (bits as usize))
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/power_of_two_impl.rs]: done

File: src/fixeduint/power_of_two_impl.rs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            // non-power-of-two inputs correctly.
+            let m_one = self - Self::one();
+            let leading = ConstPrimInt::leading_zeros(m_one);
+            let bits = Self::BIT_SIZE as u32 - leading;
+
+            // Check for overflow
+            if bits >= Self::BIT_SIZE as u32 {
+                return None;
+            }
+            Some(Self::one() << (bits as usize))
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/power_of_two_impl.rs]: nah
